miso-web/src/main/webapp/scripts/sample_hot.js

```diff
@@ -793,10 +793,11 @@ Sample.hot = {
     return sample;
   },
   
-  saveOneSample: function (data, index, messages, numberToSave) {
+  saveOneSample: function (data, index, messages, numberToSave, callback) {
     var xhr = new XMLHttpRequest();
     xhr.onreadystatechange = function() {
       if (xhr.readyState === XMLHttpRequest.DONE) {
+    	callback(); // Indicate request has completed.
```

Comment:
The callback starts sending the second sample to the server while the front end processes whether the first sample was successfully saved, so having those execute out of order shouldn't be a problem.